scripts/build.py

```diff
@@ -181,6 +181,91 @@ def _ensure_files_exist(filepaths):
             raise Exception('File %s does not exist.' % filepath)
 
 
+def _ensure_fonts_exist(filepaths):
+    """Ensures that fonts exists at the given filepaths.
+
+    Args:
+        filepaths: list(str). Wildcard paths to fonts.
+
+    Raises:
+        Exception: Raised if one or more of the files does not exist.
+    """
+    for font_wildcard in filepaths:
+        font_paths = glob.glob(font_wildcard)
+        _ensure_files_exist(font_paths)
+
+
+def get_file_count(directory_path):
+    """Count total number of file directory, subtracting ignored files.
+
+    Args:
+        directory_path: str. Directory to be walked.
+
+    Returns:
+        total_file_count: int. Total number of files minus ignored files.
+    """
+    total_file_count = 0
+    #pylint: disable=unused-variable
+    for root, dirs, files in os.walk(directory_path):
+        #pylint: enable=unused-variable
+        total_file_count += len(files)
+        for filename in files:
+            # Ignore files with certain extensions.
+            if any(filename.endswith(p)
+                   for p in FILE_EXTENSIONS_TO_IGNORE):
+                total_file_count -= 1
+    if directory_path == ASSETS_DEV_DIR:
+        # Add 1 to account for the added hashes.js.
+        total_file_count += 1
+    return total_file_count
+
+
+def _compare_file_count(source_path, target_path):
+    """Ensure that two dir's file counts matches.
+
+    Args:
+       source_path: str. Source of staging files.
+       target_path: str. Final filepath of built files.
+           the processed file.
+
+    Raises:
+        Exception: Raised if both dir do not have the same file count.
+    """
+    source_dir_file_count = get_file_count(source_path)
+    target_dir_file_count = get_file_count(target_path)
+    if source_dir_file_count != target_dir_file_count:
+        raise Exception(
```

Comment:
I think we should create custom Exceptions. So maybe define smth like at the start of the file: 
```python
class FileCountException(Exception):
    pass
``` 
and then raise it here. We should do that for all exceptions used I think.